// prettyput/lib/put.js
// Generated by CoffeeScript 1.8.0
module.exports = {
  e: function(err) {
    if (err != null) {
      return this.p(err);
    }
  },
  p: function(obj) {
    if ((typeof console !== "undefined" && console !== null) && (console.log != null)) {
      return console.log(obj);
    }
  },
  o: function(obj, depth, indent_size) {
    var key;
    if (depth == null) {
      depth = 0;
    }
    if (indent_size == null) {
      indent_size = 2;
    }
    if (depth === 0) {
      depth += 1;
      this.p('{');
    }
    for (key in obj) {
      if (typeof obj[key] === 'object') {
        this.p(this._indent(depth, indent_size) + key + ':{ ');
        this.o(obj[key], depth + 1);
        this.p(this._indent(depth, indent_size) + '}');
      } else {
        this.p(this._indent(depth, indent_size) + key + ':' + obj[key]);
      }
    }
    if (depth === 1) {
      return this.p('}');
    }
  },
  a: function(arr) {
    var i, v, _i, _len, _results;
    _results = [];
    for (i = _i = 0, _len = arr.length; _i < _len; i = ++_i) {
      v = arr[i];
      _results.push(this.p(i + ' -> ' + v));
    }
    return _results;
  },
  w: function(obj) {
    return this.p('-> ' + obj + ' <-');
  },
  n: function() {
    return this.p('');
  },
  pp: function(arr) {
    var v, _i, _len, _results;
    _results = [];
    for (_i = 0, _len = arr.length; _i < _len; _i++) {
      v = arr[_i];
      _results.push(this.p(v));
    }
    return _results;
  },
  _indent: function(depth, size) {
    var i, res, tmp, _i, _j;
    res = '';
    for (i = _i = 0; 0 <= size ? _i < size : _i > size; i = 0 <= size ? ++_i : --_i) {
      res += ' ';
    }
    if (depth === 0) {
      return '';
    } else {
      tmp = '';
      for (i = _j = 0; 0 <= depth ? _j < depth : _j > depth; i = 0 <= depth ? ++_j : --_j) {
        tmp += res;
      }
      return tmp;
    }
  }
};
